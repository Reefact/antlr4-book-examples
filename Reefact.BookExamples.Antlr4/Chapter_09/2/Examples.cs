#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_09._2 {

    [UseReporter(typeof(BeyondCompareReporter), typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        public void verbose_listener() {
            // Setup
            AntlrInputStream inputStream   = AntlrInputStreamReader.Read("class T T { int : }");
            GRun             grun          = GRun.Read(inputStream);
            VerboseListener  errorListener = new();
            // Exercise
            string output = grun.GetOutput(errorListener);
            // Verify
            Approvals.Verify(output);
        }

        [Fact]
        public void underline_listener() {
            // Setup
            AntlrInputStream  inputStream   = AntlrInputStreamReader.Read("underline_listener.simple", 9, 2);
            GRun              grun          = GRun.Read(inputStream);
            UnderlineListener errorListener = new();
            // Exercise
            string output = grun.GetOutput(errorListener);
            // Verify
            Approvals.Verify(output);
        }

    }

}