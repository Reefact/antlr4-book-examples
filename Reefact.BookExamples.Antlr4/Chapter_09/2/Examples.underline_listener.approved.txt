line 1:8 extraneous input 'XYZ' expecting '{'
class T XYZ {
        ^^^
line 2:4 no viable alternative at input 'int;'
int ;
    ^
class T